lib/membrane_flv_plugin/muxer.ex

```diff
@@ -147,23 +170,20 @@ defmodule Membrane.FLV.Muxer do
     |> prepare_to_send(state)
   end
 
-  @impl true
-  def handle_stream_format(Pad.ref(type, _id) = _pad, stream_format, _ctx, _state),
-    do:
-      raise(
-        "Stream format '#{inspect(stream_format)}' is not supported for stream type #{inspect(type)}"
-      )
+  defp handle_queue_item({Pad.ref(type, _id), {:stream_format, stream_format}}, _state) do
+    raise """
+    Stream format '#{inspect(stream_format)}' is not supported for stream type #{inspect(type)}"
+    """
+  end
 
-  @impl true
-  def handle_end_of_stream(pad, ctx, state) do
-    # Check if there are any input pads that didn't eos. If not, send end of stream on output
-    state = Map.update!(state, :last_dts, &Map.delete(&1, pad))
+  defp handle_queue_item({pad, :end_of_stream}, state) do
+    state = Map.update!(state, :inputs_without_eos, &MapSet.delete(&1, pad))
```

Comment:
Could we add checking for pads with eos to queue?

Reply:
Gimme approve https://github.com/membraneframework/membrane_timestamp_queue/pull/6/files